tokens/src/commands.rs

```diff
@@ -492,61 +499,79 @@ fn read_allocations(
     transfer_amount: Option<u64>,
     require_lockup_heading: bool,
     raw_amount: bool,
-) -> io::Result<Vec<Allocation>> {
+) -> Result<Vec<Allocation>, Error> {
     let mut rdr = ReaderBuilder::new().trim(Trim::All).from_path(input_csv)?;
     let allocations = if let Some(amount) = transfer_amount {
-        let recipients: Vec<String> = rdr
-            .deserialize()
-            .map(|recipient| recipient.unwrap())
-            .collect();
-        recipients
-            .into_iter()
-            .map(|recipient| Allocation {
-                recipient,
-                amount,
-                lockup_date: "".to_string(),
+        rdr.deserialize()
+            .map(|recipient| {
+                let recipient: String = recipient?;
+                match Pubkey::from_str(recipient.as_str()) {
+                    Ok(..) => (),
+                    Err(err) => return Err(Error::BadInputPubkeyError(err)),
+                }
+                Ok(Allocation {
+                    recipient,
+                    amount,
+                    lockup_date: "".to_string(),
+                })
```

Comment:
We might also try/want to change field types on `Allocation` (so we don't have to parse these twice):
```
pub struct Allocation {
    pub recipient: String, // -> pub recipient: Pubkey,
    pub amount: u64,
    pub lockup_date: String, // -> pub lockup_date: Option<DateTime<Utc>>,
}
```
but maybe it's not worth the effort, wdyt ?

Reply:
I really like the idea of typing those `Allocation` fields like you suggest, so that data validation and parsing happens all at once in `read_allocations`. I can't think of any assumptions that would break... do you mind reworking this in that direction so we can see?

Reply:
Seems to be working with new types: https://github.com/solana-labs/solana/pull/33926/commits/30dfbaabae71d91c9da6907e82e465d22fee3fb1